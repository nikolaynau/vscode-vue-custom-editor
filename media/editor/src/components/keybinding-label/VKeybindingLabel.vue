<script setup lang="ts">
import { computed, toRefs } from 'vue';

export interface Props {
  value?: string;
  separator?: string;
}

const props = withDefaults(defineProps<Props>(), {
  value: undefined,
  separator: '+'
});

const { value, separator } = toRefs(props);

const keyParts = computed(() =>
  (value.value || '').split(separator.value).filter(item => !!item)
);
</script>

<template>
  <div v-if="value" class="v-keybinding-label" :title="value">
    <template v-for="(part, index) in keyParts" :key="index">
      <span v-if="index > 0" class="v-keybinding-label__separator">
        {{ separator }}
      </span>
      <span class="v-keybinding-label__key">{{ part }}</span>
    </template>
  </div>
</template>

<style>
@import 'keybinding-label';
</style>
