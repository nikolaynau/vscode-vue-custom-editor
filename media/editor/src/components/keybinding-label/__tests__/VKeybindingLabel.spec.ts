import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import VKeybindingLabel from '../VKeybindingLabel.vue';

describe('VKeybindingLabel', () => {
  it('render default props', () => {
    const wrapper = mount(VKeybindingLabel);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('value prop', () => {
    const wrapper = mount(VKeybindingLabel, { props: { value: 'Ctrl+Enter' } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('separator prop', () => {
    const wrapper = mount(VKeybindingLabel, {
      props: { value: 'Ctrl Enter', separator: ' ' }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
