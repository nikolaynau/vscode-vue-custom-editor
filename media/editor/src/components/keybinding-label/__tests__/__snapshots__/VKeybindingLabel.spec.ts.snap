// Vitest Snapshot v1

exports[`VKeybindingLabel > render default props 1`] = `"<!--v-if-->"`;

exports[`VKeybindingLabel > separator prop 1`] = `
"<div class=\\"v-keybinding-label\\" title=\\"Ctrl Enter\\">
  <!--v-if--><span class=\\"v-keybinding-label__key\\">Ctrl</span><span class=\\"v-keybinding-label__separator\\"> </span><span class=\\"v-keybinding-label__key\\">Enter</span>
</div>"
`;

exports[`VKeybindingLabel > value prop 1`] = `
"<div class=\\"v-keybinding-label\\" title=\\"Ctrl+Enter\\">
  <!--v-if--><span class=\\"v-keybinding-label__key\\">Ctrl</span><span class=\\"v-keybinding-label__separator\\">+</span><span class=\\"v-keybinding-label__key\\">Enter</span>
</div>"
`;
